lxd/device/gpu_physical.go

```diff
@@ -120,42 +120,44 @@ func (d *gpuPhysical) startContainer() (*deviceConfig.RunConfig, error) {
 		found = true
 
 		// Setup DRM unix-char devices if present.
-		if gpu.DRM.CardName != "" && gpu.DRM.CardDevice != "" && shared.PathExists(filepath.Join(gpuDRIDevPath, gpu.DRM.CardName)) {
-			path := filepath.Join(gpuDRIDevPath, gpu.DRM.CardName)
-			major, minor, err := d.deviceNumStringToUint32(gpu.DRM.CardDevice)
-			if err != nil {
-				return nil, err
-			}
+		if gpu.DRM != nil {
```

Comment:
Ups, looks like this was introduced by me in https://github.com/lxc/lxd/pull/11799. I think changing the `gpuSelected` method to the following would fix it for all gpu types, since the case were the graphics card is there but without being recognized on the host is missing (e.g. no nvidia drivers installed on the system).

This also makes the logic "more clear":

```go
func gpuSelected(device config.Device, gpu api.ResourcesGPUCard) bool {
	if device["vendorid"] != "" && gpu.VendorID == device["vendorid"] {
		return true
	}
	if device["pci"] != "" && gpu.PCIAddress == device["pci"] {
		return true
	}
	if device["productid"] != "" && gpu.ProductID == device["productid"] {
		return true
	}
	if device["id"] != "" && gpu.DRM != nil && fmt.Sprintf("%d", gpu.DRM.ID) == device["id"] {
		return true
	}
	return false
}
```

Reply:
I think we should do both, as the not-nil check is present for the other fields and is just good defensive coding.

Please can you open a PR for the suggestion you made here, as it sounds good to me as well.

Reply:
Also can you confirm whether this would "break" passing through the GPU device from the Github runners (not really desirable) because it lacks DRM field?

Reply:
So the github test runners have a `/dev/dri/card0` device and `lxc config device add "${ctName}" gpu-default gpu` will need to work.

Reply:
This seems to work https://github.com/lxc/lxd/blob/master/test/suites/container_devices_gpu.sh#L16

Reply:
I understand if the host does not have the right drivers installed, the DRM info is missing. So you cannot use the `id` field when adding the GPU:

`lxc config device add c1 gpu-default gpu id=1`

Instead to address just the device, use the `pci={addr}` selector:

`lxc config device add c1 gpu-default gpu pci={addr}`.

From what I saw this was the original problem of https://github.com/lxc/lxd/issues/11442 because it added all the devices to the container even though none of them was selected.